anndata/core/aligned_mapping.py

```diff
@@ -329,24 +363,25 @@ def dim(self) -> str:
         return self._dimnames[self._axis]
 
 
-class PairwiseArrays(AlignedActualMixin, PairwiseArraysBase):
+class _PairwiseArraysActual(AlignedActual, PairwiseArrays):
     def __init__(
-        self, parent: "AnnData", axis: int, vals: Optional[Mapping] = None
+        self,
+        parent: "anndata.AnnData",
+        axis: int,
+        vals: Mapping[str, V] = MappingProxyType({}),
     ):
         self._parent = parent
-        if axis not in (0, 1):
-            raise ValueError()
+        assert axis in (0, 1)
```

Comment:
Why this change?

Reply:
Again: `asserts` are for us misusing internal interfaces or violating assumptions. “Not your fault user, but ours. Sorry and please file a bug”

At the moment this is an internal interface. We can change it back (with an actual helpful message) if it becomes a public one.

Reply:
answered above